### PR TITLE
ENH/TST: Windows compatibility, Improved interface with optional packages

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #88 and #90

- Added Github Actions testing for windows
- Moved `OMMBV`, `aacgmv2`, and `apexpy` to optional packages
- Created `package_check`, a decorator function that can be used to quickly assess whether the optional packages are actually installed and bypass so that `pysatMissions` can still be imported and utilized for core operations.
- Added pytest skip statements to the unit tests for `aacgmv2` and `apexpy` if these optional packages are not installed.  This can be determined by running `pytest -vs` or inspection of coveralls.
- Reduced dependency on `OMMBV` by including some local vector calculation functions in `methods.spacecraft`. Note that the only remaining dependency on OMMBV is in the deprecated `missions_ephem` instrument, which will be removed in v0.4.0.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Via github actions and by local pytest runs after uninstalling OMMBV and apexpy

**Test Configuration**:
* Mac OS X Big Sur
* Python 3.8.11

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysatMissions/tests/test_methods_spacecraft.py [hunk 1]
```diff
@@ -79,7 +79,6 @@ def test_calculate_ecef_velocity(self):
         self.eval_targets(targets)
         return
 
-    @pytest.mark.skipif(("OMMBV" not in dir()), reason="OMMBV is not installed")
     def test_add_ram_pointing_sc_attitude_vectors(self):
```
Comment (jklenzing) [inline pysatMissions/tests/test_methods_spacecraft.py:82]: This test no longer depends on OMMBV

**Comment by jklenzing** [comment]
Inspecting coveralls, OMMBV successfully installs for both linux and windows, apexpy is not installing on windows.  However, the bypass is working as intended.

**Comment by jklenzing** [comment]
TODO: compare output calculations here with develop.  Needs working OMMBV.

**Comment by jklenzing** [comment]
I've yet to run into issues with aacgm, but we could move that one to optional as well so that fortran is not required for basic operations.

**Comment by jklenzing** [comment]
Output values for transformed vector in `test_methods_scacecraft.py::TestBasics::test_project_ecef_vector_onto_sc`` match that in `develop`.  This routine runs all the changed functions in order to map one vector onto another after normalizing.

**Comment by jklenzing** [comment]
This one is ready to go.  I'll probably open another branch downstream to add the mac and nep29 tests since I think the changes here will also improve the Github Actions issues we were running into there, but the pull is big enough already.

**Comment by jklenzing** [comment]
pinging @rstoneback

**Review by rstoneback — APPROVED** [review]
Thanks @jklenzing. Apologies for my delay.

**Pull request merged**